### PR TITLE
Private/devidas/attach node spinner

### PR DESCRIPTION
**Pull request by devidask27**
Testing
**Attaching 3 workers**
Screenshot 2022-01-20 at 4 43 06 PM
**Attaching 3 workers which are already connected to cluster**
Screenshot 2022-01-20 at 4 44 23 PM
**Attaching 3 workers out of three one is already connected to cluster**
Screenshot 2022-01-20 at 4 46 23 PM
**Attaching 3 workers out of three 2 are already connected to cluster**
Screenshot 2022-01-20 at 4 47 22 PM
**Attaching single worker node**
Screenshot 2022-01-20 at 4 49 12 PM
**Attaching single master node**
Screenshot 2022-01-20 at 5 40 22 PM
**Attaching 3 node as master which are already connected to cluster**
Screenshot 2022-01-20 at 5 04 56 PM
**Attaching one master two workers**
Screenshot 2022-01-20 at 5 08 33 PM
**When cluster is not ready**
Screenshot 2022-01-20 at 5 09 25 PM

### CODE REVIEW

File: cmd/attachNode.go [hunk 1]
```diff
@@ -118,39 +118,67 @@ func attachNodeRun(cmd *cobra.Command, args []string) {
...
+					}
+					zap.S().Info("Encountered an error while attaching worker node to a Kubernetes cluster : ", err1)
+				} else {
+					if err := c.Segment.SendEvent("Attaching-node", auth, "Worker node attached", ""); err != nil {
+						zap.S().Errorf("Unable to send Segment event for attach node. Error: %s", err.Error())
+					}
+					zap.S().Infof("Worker node(s) %v attached to cluster", wokerids)
 				}
-				zap.S().Infof("Worker node(s) %v attached to cluster", worker_hostIds)
 			}
```
Comment (nikhilbandi) [inline cmd/attachNode.go:152]: Shouldn't we have else condition where we print console message and log saying "no worker node available to attach to the cluster"?

File: cmd/attachNode.go [hunk 2]
```diff
@@ -118,39 +118,67 @@ func attachNodeRun(cmd *cobra.Command, args []string) {
...
+					}
+					zap.S().Info("Encountered an error while attaching master node to a Kubernetes cluster : ", err1)
+				} else {
+					if err := c.Segment.SendEvent("Attaching-node", auth, "Master node attached", ""); err != nil {
+						zap.S().Errorf("Unable to send Segment event for attach node. Error: %s", err.Error())
+					}
+					zap.S().Infof("Master node(s) %v attached to cluster", masterids)
 				}
-				zap.S().Infof("Master node(s) %v attached to cluster", master_hostIds)
 			}
```
Comment (nikhilbandi) [inline cmd/attachNode.go:180]: Same as above, just for Master nodes. Shouldn't the relevant message be printed in the else condition?

File: cmd/attachNode.go [hunk 3]
```diff
@@ -189,3 +217,15 @@ func cluster_Status(exec cmdexec.Executor, fqdn string, token string, projectID
 	zap.S().Debug("Cluster status is : ", status)
 	return status
 }
+
```
Comment (nikhilbandi) [inline cmd/attachNode.go:220]: Add comment "//Check if the node being attached is already attached to any cluster."

**Review by nikhilbandi — CHANGES_REQUESTED** [review]
Please take a look at the comments mentioned.

File: cmd/attachNode.go (hunk 1 above)
Comment (devidask27) [reply cmd/attachNode.go:152]: No Nikhil, we are first printing if it is attached to any cluster or not. So there is no node means that are already connected.

Comment (devidask27) [reply cmd/attachNode.go:152]: Done

File: cmd/attachNode.go (hunk 2 above)
Comment (devidask27) [reply cmd/attachNode.go:180]: Done

File: cmd/attachNode.go (hunk 3 above)
Comment (devidask27) [reply cmd/attachNode.go:220]: Done

**Review by nikhilbandi — APPROVED** [review]
Approving.

**Pull request merged**